### ISSUE TITLE
A player who dies from the Snare spell doesn't produce a grave on death

### ISSUE THREAD

**Issue by Not-February** [Electroblob77/Wizardry#359]
**Minecraft version:** 1.12.2
**Wizardry version:** 4.2.8
**Environment:** Dedicated Server

**Issue details:** A player on my private server discovered accidentally that if you die to the Snare spell, the death is triggered and the player is sent to the respawn screen... but there is no death message on the server's end, or even a death _event_ that occurs, because the grave mod we're using flat out didn't even register that the player died.

They respawned and their entire inventory was gone(presumably because of the grave mod), but no grave/relevant items or information were generated at all. I wasn't able to restore their inventory from a backup command via the grave mod because there wasn't a grave to restore.

The player in question was streaming at the time and managed to get footage of this happening, here's a clip(they were apparently trying to trial-and-error a spell and it spawned the Snare traps under them):

https://clips.twitch.tv/CautiousJollyWasabiBatChest

Important to note is the lack of any triggered death message in chat upon their death. It wasn't in the server log when I checked, wasn't picked up by a Discord chat relay when other deaths had been, it's like it didn't happen at all somehow - except it did.

I don't know for certain whether this is relevant specifically to the Snare or whether it affects similar "trap"-type spells, but I would say there's a good chance that it does. I can bring this to the grave mod's issue tracker if it's on their end and not yours.

**Other mods involved:** Corail Tombstone

**Comment by Electroblob77** [comment]
Actually, I suspect this is specific to snare only, because it's a block whereas the other trap spells are entities. Snare has had some syncing issues in the past, what I suspect is happening from your description is that the player is being killed on the client side but not on the server, hence the server doesn't know they've died and doesn't broadcast a message or give the items back. I'll look into this, but you may want to disable the snare spell in the meantime.

Edit: I wonder if this has anything to do with #15?

**Comment by Not-February** [comment]
I have disabled the Snare spell in the config since then, but unfortunately, I've found that despite this the snares can still be placed in-world by wayward attempts at identifying Earth-affiliated spells, so it's still a little dicey.

Good to know I can safely re-enable the other "trap" type spells, though.

I don't know exactly what you're referring to when mentioning #15 as being related; it's possible that the player who found this issue did install OptiFine on their end, but I would have to ask to double check. Or is it related moreso in the thought that the issue here might be caused by a quirk when interacting with another mod?

I don't have OptiFine or BetterFPS included in my pack by default, and I'm not sure what other mods could interfere right off the bat - here's a [Modlister link](https://www.modlister.com/user/febilian/Beans%20Sold%20Separately) to what I've got installed if you think it would help narrow down any possible common factors.
